reconcile/test/test_terraform_vpc_resources.py

```diff
@@ -0,0 +1,127 @@
+import logging
+
+import pytest
+
+from reconcile.gql_definitions.common.app_interface_vault_settings import (
+    AppInterfaceSettingsV1,
+)
+from reconcile.gql_definitions.terraform_vpc_resources.vpc_resources_aws_accounts import (
+    AWSAccountV1,
+    AWSTerraformStateIntegrationsV1,
+    TerraformStateAWSV1,
+    VPCResourcesAWSAccountsQueryData,
+)
+from reconcile.status import ExitCodes
+from reconcile.terraform_vpc_resources import (
+    TerraformVpcResources,
+    TerraformVpcResourcesParams,
+)
+
+
+@pytest.fixture
+def query_data():
+    return VPCResourcesAWSAccountsQueryData(
+        accounts=[
+            AWSAccountV1(
+                name="some-account",
+                terraformState=TerraformStateAWSV1(
+                    integrations=[
+                        AWSTerraformStateIntegrationsV1(
+                            key="some-key",
+                            integration="some-integration",
+                        ),
+                    ]
+                ),
+            ),
+            AWSAccountV1(
+                name="some-other-account",
+                terraformState=TerraformStateAWSV1(
+                    integrations=[
+                        AWSTerraformStateIntegrationsV1(
+                            key="terraform-vpc-resouces.tf.json",
+                            integration="terraform-vpc-resources",
+                        ),
+                    ]
+                ),
+            ),
+        ]
+    )
+
+
+@pytest.fixture
+def mock_gql(mocker):
```

Comment:
We have a global fixture for the gql api - might come in handy https://github.com/app-sre/qontract-reconcile/blob/b14feca7affff061773cf6cf712da2de3fec488a/reconcile/test/conftest.py#L146

Examples of how this can be used https://github.com/app-sre/qontract-reconcile/blob/b14feca7affff061773cf6cf712da2de3fec488a/reconcile/test/test_typed_queries/test_jira_settings.py#L37-L49

Reply:
I'm only seeing this being used in `typed_queries` and in `terraform-cloudflare-users`. And in `terraform-cloudflare-users` is created just to be used as the return of a `mocker.patch`.

I can see the value if we use api as dependency of the integration like you commented below. But that would be something new codebase wise, as far as I know.

Reply:
Yes this makes sense if you use DI. If you use mock.patch instead, then you dont need that fixture.